worker/docker_image_manager.py

```diff
@@ -0,0 +1,140 @@
+import threading
+import os
+import time
+import traceback
+import logging
+import json
+
+import file_util
+from formatting import size_str
+
+logger = logging.getLogger(__name__)
+
+
+class DockerImageManager(object):
+    """
+    Manages the Docker images available on the worker, and ensures that they
+    do not fill up the disk.
+
+    The last-used dates are only tracked for images on which
+    `DockerImageManager.touch_image()` has been called. All other images will
+    be left untouched by the cleanup thread.
+
+    DependencyManager should be instantiated before DockerImageManager, to
+    ensure that the work directory already exists.
+    """
+    STATE_FILENAME = 'images-state.json'
+
+    def __init__(self, docker, work_dir, min_free_bytes=1e9,
```

Comment:
10GB and config file parameter sound good, and uniform cleanup logic would be great.

I think we might need a bit more than just specifying 10GB to each cleanup thread though: when the disk has less than 10 GB left, how do they decide whether to delete an image or a bundle? That's why I was hoping for some way to query for the amount of disk space used by all the Docker images, so that we could just independent limits on both bundles and images, such that the sum of the limits is less than the size of the disk.

But since that was impossible, I decided to use a disk free parameter, which means that while the dependency manager independently caps the amount of disk space used by bundles, the image manager ensures that images never fill up the remainder of the disk completely.